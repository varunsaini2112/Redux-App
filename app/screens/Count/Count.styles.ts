import { StyleSheet } from "react-native";
import { Padding, colors } from "../../config/appStyles";

const styles = StyleSheet.create({
	mainContainer: {
		flex: 1,
		justifyContent: "center",
		alignItems: "center"
	},
	button: {
		padding: Padding.xl,
		backgroundColor: colors.primary,
		margin: Padding.xl,
		borderRadius: Padding.xl,
		alignItems: "center"
	}
});

export default styles;
