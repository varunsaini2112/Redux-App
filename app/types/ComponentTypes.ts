import { RootStackParamList } from ".";

type LinkType = {
	value: string;
	href: keyof RootStackParamList;
};

type LinkProps = LinkType & {
	onPress: (screenName: keyof RootStackParamList) => void;
};

export type { LinkType, LinkProps };
