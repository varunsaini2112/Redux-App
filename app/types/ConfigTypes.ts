type RootStackParamList = {
	Home: undefined;
	Count: undefined;
	Login: undefined;
	ToDo: undefined;
};

export type { RootStackParamList };
