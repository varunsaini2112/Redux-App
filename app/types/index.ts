export * from "./ConfigTypes";

export * from "./ComponentTypes";
