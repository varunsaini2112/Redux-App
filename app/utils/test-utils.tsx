import React, { PropsWithChildren, ReactElement } from "react";
import { Provider } from "react-redux";
import { PreloadedState } from "@reduxjs/toolkit";
import { RenderOptions, render } from "@testing-library/react-native";

import { AppStore, RootState, setupStore } from "../redux/store";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
	preloadedState?: PreloadedState<RootState>;
	store?: AppStore;
}

export function renderWithProviders(
	ui: ReactElement,
	{
		preloadedState = {},
		// Automatically create a store instance if no store was passed in
		store = setupStore(preloadedState),
		...renderOptions
	}: ExtendedRenderOptions = {}
) {
	function Wrapper({ children }: PropsWithChildren<{}>): ReactElement {
		return <Provider store={store}>{children}</Provider>;
	}
	return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
