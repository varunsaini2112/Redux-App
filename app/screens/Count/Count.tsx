import { FC, useState } from "react";
import { Pressable, Text, View } from "react-native";
import { StackScreenProps } from "@react-navigation/stack";

import styles from "./Count.styles";
import { H1 } from "../../components";
import { RootStackParamList } from "../../types";

const Count: FC<StackScreenProps<RootStackParamList, "Count">> = () => {
	const [count, setCount] = useState(0);

	const increment = () => setCount((count) => count + 1);
	const decrement = () => setCount((count) => count - 1);
	const reset = () => setCount(0);

	return (
		<View style={styles.mainContainer}>
			<H1>{count}</H1>
			<View>
				<Pressable onPress={increment} style={styles.button}>
					<Text>Increment</Text>
				</Pressable>
				<Pressable onPress={decrement} style={styles.button}>
					<Text>Decrement</Text>
				</Pressable>
				<Pressable onPress={reset} style={styles.button}>
					<Text>Reset</Text>
				</Pressable>
			</View>
		</View>
	);
};

export default Count;
