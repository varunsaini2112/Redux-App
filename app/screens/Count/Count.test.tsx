import { fireEvent, screen } from "@testing-library/react-native";

import Count from "./Count";
import { StackNavigationProp } from "@react-navigation/stack";
import { RootStackParamList } from "../../types";
import { RouteProp } from "@react-navigation/native";
import { renderWithProviders } from "../../utils/test-utils";

const navigation = {
	navigate: () => {}
} as StackNavigationProp<RootStackParamList, "Count">;
const route = {} as RouteProp<RootStackParamList, "Count">;

describe("App", () => {
	it("should match snapshots", () => {
		const snapshot = renderWithProviders(
			<Count navigation={navigation} route={route} />
		);

		expect(snapshot).toMatchSnapshot();
	});

	it("Count renders 0 initially", () => {
		renderWithProviders(<Count navigation={navigation} route={route} />);

		expect(screen.getByText("0")).toBeOnTheScreen();
	});

	it("Count should increase on press increment button", () => {
		renderWithProviders(<Count navigation={navigation} route={route} />);

		const incrementButton = screen.getByText("Increment");
		fireEvent.press(incrementButton);

		expect(screen.getByText("1")).toBeOnTheScreen();
	});

	it("Count should decrease on press decrement button", () => {
		renderWithProviders(<Count navigation={navigation} route={route} />);

		const incrementButton = screen.getByText("Decrement");
		fireEvent.press(incrementButton);

		expect(screen.getByText("-1")).toBeOnTheScreen();
	});

	it("Count should increase on press increment button", () => {
		renderWithProviders(<Count navigation={navigation} route={route} />);

		const incrementButton = screen.getByText("Increment");
		const resetButton = screen.getByText("Reset");
		fireEvent.press(incrementButton);
		fireEvent.press(incrementButton);
		fireEvent.press(incrementButton);

		expect(screen.getByText("3")).toBeOnTheScreen();

		fireEvent.press(resetButton);
		expect(screen.getByText("0")).toBeOnTheScreen();
	});
});
