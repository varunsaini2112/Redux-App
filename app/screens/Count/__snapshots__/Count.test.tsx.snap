// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshots 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#132631",
        "fontSize": 24,
        "fontWeight": "600",
      }
    }
  >
    0
  </Text>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#D80064",
          "borderRadius": 20,
          "margin": 20,
          "padding": 20,
        }
      }
    >
      <Text>
        Increment
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#D80064",
          "borderRadius": 20,
          "margin": 20,
          "padding": 20,
        }
      }
    >
      <Text>
        Decrement
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#D80064",
          "borderRadius": 20,
          "margin": 20,
          "padding": 20,
        }
      }
    >
      <Text>
        Reset
      </Text>
    </View>
  </View>
</View>
`;
